e2e/test/scenarios/models/reproductions/31406-model-link-formatting.cy.spec.js

```diff
@@ -0,0 +1,75 @@
+import { modal, popover, restore } from "e2e/support/helpers";
+
+describe.skip("issue 31406", () => {
+  beforeEach(() => {
+    restore();
+    cy.signInAsAdmin();
+
+    cy.visit("/");
+
+    createPeopleModel();
+  });
+
+  it("should persist link text formatting settings for link column (metabase#31406)", () => {
+    cy.findByTestId("TableInteractive-root").findByText("Email").click();
+
+    popover().within(() => {
+      cy.icon("gear").click();
+    });
+
+    popover().within(() => {
+      cy.findByPlaceholderText("Link to {{bird_id}}")
+        .type("Link to {{name}}", {
+          parseSpecialCharSequences: false,
+        })
+        .blur();
+    });
+
+    expect(
+      cy
+        .findByTestId("TableInteractive-root")
+        .findByText("Link to Hudson Borer"),
+    );
+
+    cy.findByTestId("dataset-edit-bar").findByText("Save").click();
+
+    modal().within(() => {
+      // make sure it is a model and hasn't been converted to a question
+      cy.findByText("Save model");
+
+      cy.findByText("Save").click();
+    });
+
+    modal().within(() => {
+      cy.findByText("Saved! Add this to a dashboard?");
+
+      cy.findByText("Not now").click();
+    });
+  });
+});
+
+function createPeopleModel() {
+  cy.intercept("POST", "/api/card").as("createCard");
+
+  cy.findByText("New").click();
+  cy.findByText("Model").click();
+
+  cy.findByText("Use the notebook editor").click();
+
+  popover().within(() => {
+    cy.findByText("Sample Database").click();
+    cy.findByText("People").click();
+  });
+
+  cy.findByText("Save").click();
+
+  modal().within(() => {
+    cy.findByLabelText("Name").type("{selectall}People Model");
+
+    cy.findByText("Save").click();
+  });
+
+  cy.wait("@createCard");
+
+  cy.get(".RunButton .Icon-play").click();
```

Comment:
```suggestion
const questionDetails = {
  query: {
    "source-table": PEOPLE_ID,
    dataset: true,
    limit: 2, // speeds test up significantly without affecting our confidence in this test
  },
};

cy.createQuestion(questionDetails, { visitQuestion: true });
```

Literally all of this can be replaced with the API call.
The added benefit is that all of the intercepts and waits are set up for you for free under the hood :)

Reply:
Awesome!
Also, forgot to mention you can add `name` to the question details object if you need to assert on it later. If not, we will set the fallback value for `name` to "test question".

The reason I like to use `const questionDetails =...` is that I don't have to rely on hard coded data later in test. I can do something like `selector(foo).should("have.text", questionDetails.name)`